### PR TITLE
Allow filename input directly from LoadSave window

### PR DESCRIPTION
**Pull request by AaronVanGeffen**
This PR adds a filename input box to the bottom of the LoadSave window, allowing players to input the filename directly. Players can also use the enter/return key to confirm the new filename.

~~To make room for this input box, I have removed the 'Use system file browser' button. I figure it's obsolete, since we made it configurable years ago. (The button preceded the setting, and was just never removed.)~~

This basically makes the 'New file' button unnecessary/obsolete, but I have different ideas for the toolbar. I first want to see if/how this idea lands, so I will leave it for now.

To do:
- [x] Get consensus on whether or not we want this.
- [x] Add keyboard support (enter/esc) for confirmation boxes.
- [x] Re-introduce 'Use system file browser' button per request.

Screenshot 2024-12-14 at 17 33 04

### CODE REVIEW

**Comment by 733737** [comment]
I definitely like the idea. Maybe you could put a new folder icon to the left of "Filename:" which would replace the new folder button at the top. Similarly, the default and up buttons could be moved into the file list at the top instead of being above it.

**Comment by AaronVanGeffen** [comment]
Yes, I was thinking of replacing those buttons with icons, too. I prefer to leave the buttons/toolbar as-is in this PR, however.

File: src/openrct2-ui/input/InputManager.cpp [hunk 1]
```diff
@@ -209,6 +209,7 @@ void InputManager::Process(const InputEvent& e)
 
         if (e.DeviceKind == InputDeviceKind::Keyboard)
         {
+            // TODO: replace with event
```
Comment (AaronVanGeffen) [inline src/openrct2-ui/input/InputManager.cpp:212]: In OpenLoco, we've added an `OnKeyDown` event. I looked into adding one in OpenRCT2, however, I think we really need to sort out our window code before doing so...

**Comment by mrmbernardi** [comment]
I think the button should read `Save` instead of `OK`. Just to make it even more obvious what it's going to do.

**Comment by AaronVanGeffen** [comment]
I've changed the button caption to read 'Save' as suggested:
Screenshot 2024-12-29 at 16 03 18

As with the filename label, I've made it so that the button width is inferred. This means the button is wider in Dutch, for example:
Screenshot 2024-12-29 at 16 04 36

**Review by tupaschoal — APPROVED** [review]
I tested it briefly and it looks great

**Pull request merged**